### ISSUE TITLE
Registry Passwords don't work after migrating harbor-db

### ISSUE THREAD

**Issue by saivd** [goharbor/harbor#21282]
Hello!
I'm currently experiencing an issue concerning the migration of the postgresql database from one harbor instance to another.
While all of the data has seemingly been migrated correctly, it is not possible to reach the stored endpoints for repositories.
After reentering the password for one of the repos, it started working again, which leads me to the conclusion that each password of the stored credentials is somehow corrupted.
This is a problem, because a bunch of said passwords are not known to me.

Additional Information:

- I migrated the database using pg_dumpall on the existing db and psql -U postgres -f [dumpeddb].sql in the harbor-db container of the new installation.
- Both harbor instances have been installed with the same DB-password (at least according to the harbor.yml of the existing installation)

I would be grateful for any ideas/clues on what might be the issue here. Again, all of the data seems to have been migrated correctly, except for the passwords.

**Comment by saivd** [comment]
The admin who installed harbor originally didn't note down the password and can't remember if it ever changed so I took the password which is currently in the harbor.yml, hoping it hasn't changed since the initial installation.

I tried to get the initially set password by reading out the env-variable HARBOR_ADMIN_PASSWORD in the core-container, weirdly  the variable doesn't exist on the old instance of harbor (even though I can find it using the same method on the new installation) ...

**Comment by reasonerjt** [comment]
@saivd 

> it is not possible to reach the stored endpoints for repositories.

I'm not very sure what you mean by "stored endpoints for repositories"

But generally speaking, some data in the database are encrypted via AES256, and you need to provide a key to decrypt the data before using them.  The key by default is in the file "/etc/core/key" in the harbor-core container.  
So it seems after the migration the "encryption key" in the new Harbor instance is different from that in the old Harbor instance.  Depending on how you installed Harbor, you can find the key in the secret or on the docker host you installed Harbor.  

But given you have updated some of these encrypted data in the new Harbor instance (you said you "reentered" passwords), after you update the key these new data will not decrypted b/c they have been encrypted using the new key when you "reentered" them.

**Comment by saivd** [comment]
Thanks for the reply!
With "stored endpoints for repositories" I mean the endpoints for registries, which have been migrated from the existing harbor to  the new instance. I am unabled to reach these registries, unless I "reenter" the password for each endpoint, then it works fine.
I have checked the keys in /etc/core/key and they are different indeed (same goes for the RSA private key)
Using the existing keys, I was able to read out stored passwords from the database.

Thanks alot, the issue is resolved!
